platform/machine/azure/cluster.go

```diff
@@ -85,8 +85,10 @@ func (ac *cluster) NewMachine(userdata *conf.UserData) (platform.Machine, error)
 
 func (ac *cluster) Destroy() {
 	ac.BaseCluster.Destroy()
-	if e := ac.flight.api.TerminateResourceGroup(ac.ResourceGroup); e != nil {
-		plog.Errorf("Deleting resource group %v: %v", ac.ResourceGroup, e)
+	if ac.ResourceGroup != ac.flight.ImageResourceGroup {
```

Comment:
Can you add a comment here explaining what the flight.ImageResourceGroup is and why the conditional makes sense? This probably should be a function documentation comment as well, indicating what will Destroy() destroy.

Reply:
Done